### PR TITLE
Verbose logging on object loads

### PR DESCRIPTION
**Pull request by omesser**
Add some prints for verbose logging when loading various objects to help users debug their actions.
This was missing for me when investigating a behavior in `mlem apply` while working on https://github.com/iterative/mlem.ai/issues/213 - and the output was unhelpful

## Before:
```bash
❯ mlem -v apply models/rf new_data.csv -i --it "pandas[csv]"
⏳️ Importing object from new_data.csv
2022-11-06 20:14:53,954 [DEBUG] mlem.core.hooks: Registering NumpyNumberType to DataAnalyzer
2022-11-06 20:14:53,955 [DEBUG] mlem.core.hooks: Registering NumpyNdarrayType to DataAnalyzer
2022-11-06 20:14:53,956 [DEBUG] mlem.core.hooks: Not registerting _PandasDataType to any Analyzer because it's an abstract class
2022-11-06 20:14:53,957 [DEBUG] mlem.core.hooks: Registering SeriesType to DataAnalyzer
2022-11-06 20:14:53,958 [DEBUG] mlem.core.hooks: Registering DataFrameType to DataAnalyzer
2022-11-06 20:14:53,963 [DEBUG] mlem.core.hooks: Registering PandasImport to ImportAnalyzer
⏳️ Loading model from models/rf.mlem
2022-11-06 20:14:54,239 [DEBUG] mlem.core.hooks: Registering SklearnModel to ModelAnalyzer
2022-11-06 20:14:54,240 [DEBUG] mlem.core.hooks: Registering SklearnPipelineType to ModelAnalyzer
🍏 Applying `predict` method...
```

## After:
```bash
❯ mlem -v apply models/rf new_data.csv -i --it "pandas[csv]"
⏳️ Importing object from new_data.csv
2022-11-08 16:22:31,643 [DEBUG] mlem.core.hooks: Registering NumpyNumberType to DataAnalyzer
2022-11-08 16:22:31,644 [DEBUG] mlem.core.hooks: Registering NumpyNdarrayType to DataAnalyzer
2022-11-08 16:22:31,646 [DEBUG] mlem.core.hooks: Not registerting _PandasDataType to any Analyzer because it's an abstract class
2022-11-08 16:22:31,646 [DEBUG] mlem.core.hooks: Registering SeriesType to DataAnalyzer
2022-11-08 16:22:31,647 [DEBUG] mlem.core.hooks: Registering DataFrameType to DataAnalyzer
2022-11-08 16:22:31,649 [DEBUG] mlem.core.hooks: Registering PandasImport to ImportAnalyzer
2022-11-08 16:22:31,656 [DEBUG] mlem.contrib.pandas: Loaded dataframe object, showing 'describe'
        sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
count                1.0               1.0                1.0               1.0
mean                 0.0               1.0                2.0               3.0
std                  NaN               NaN                NaN               NaN
min                  0.0               1.0                2.0               3.0
25%                  0.0               1.0                2.0               3.0
50%                  0.0               1.0                2.0               3.0
75%                  0.0               1.0                2.0               3.0
max                  0.0               1.0                2.0               3.0
⏳️ Loading model from models/rf.mlem
2022-11-08 16:22:31,669 [DEBUG] mlem.core.metadata: Loaded meta object location=Location(path='models/rf.mlem', project='/Users/oded/iterative/test_workspace/example-mlem-get-started', rev=None, uri='file:///Users/oded/iterative/test_workspace/example-mlem-get-started/models/rf.mlem', project_uri='file:///Users/oded/iterative/test_workspace/example-mlem-get-started/', fs=<fsspec.implementations.local.LocalFileSystem object at 0x102aa1c70>) params={} artifacts={'data': LocalArtifact(uri='rf', size=163651, hash='5276c1aecc8f1ad9e0e02305dd37c5dd')} requirements=Requirements(__root__=[InstallableRequirement(module='sklearn', version='1.1.3', package_name=None), InstallableRequirement(module='pandas', version='1.5.1', package_name=None), InstallableRequirement(module='numpy', version='1.23.4', package_name=None)]) model_type_cache={'methods': {'predict': {'args': [{'name': 'data', 'type_': {'columns': ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)'], 'dtypes': ['float64', 'float64', 'float64', 'float64'], 'index_cols': [], 'type': 'dataframe'}}], 'name': 'predict', 'returns': {'dtype': 'int64', 'shape': [None], 'type': 'ndarray'}}, 'predict_proba': {'args': [{'name': 'data', 'type_': {'columns': ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)'], 'dtypes': ['float64', 'float64', 'float64', 'float64'], 'index_cols': [], 'type': 'dataframe'}}], 'name': 'predict_proba', 'returns': {'dtype': 'float64', 'shape': [None, 3], 'type': 'ndarray'}}, 'sklearn_predict': {'args': [{'name': 'X', 'type_': {'columns': ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)'], 'dtypes': ['float64', 'float64', 'float64', 'float64'], 'index_cols': [], 'type': 'dataframe'}}], 'name': 'predict', 'returns': {'dtype': 'int64', 'shape': [None], 'type': 'ndarray'}}, 'sklearn_predict_proba': {'args': [{'name': 'X', 'type_': {'columns': ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)'], 'dtypes': ['float64', 'float64', 'float64', 'float64'], 'index_cols': [], 'type': 'dataframe'}}], 'name': 'predict_proba', 'returns': {'dtype': 'float64', 'shape': [None, 3], 'type': 'ndarray'}}}, 'type': 'sklearn'}
2022-11-08 16:22:32,215 [DEBUG] mlem.core.hooks: Registering SklearnModel to ModelAnalyzer
2022-11-08 16:22:32,216 [DEBUG] mlem.core.hooks: Registering SklearnPipelineType to ModelAnalyzer
🍏 Applying `predict` method...

```

## Note:
Didn't test pytorch objects - They should be just print the truncated for of the pytorch tensors afaik

### CODE REVIEW

**Comment by aguschin** [comment]
I see that failures in CI are connected with a missing s3 buckets content and changes in https://github.com/iterative/mlem-test... Not related to this PR.

@mike0sv, I assume you updated them recently?

File: mlem/contrib/pandas.py [hunk 1]
```diff
@@ -686,4 +690,8 @@ def load_obj(cls, location: Location, modifier: Optional[str], **kwargs):
         read_args = fmt.read_args or {}
         read_args.update(kwargs)
         with location.open("rb") as f:
-            return fmt.read_func(f, **read_args)
+            df = fmt.read_func(f, **read_args)
+
+            # note: should consider using head() here, more meaningful for small dfs and vectors
+            logger.debug("Loaded dataframe object, showing 'describe'\n %s", df.head())
```
Comment (aguschin) [inline mlem/contrib/pandas.py:696]: So, `head` or `describe`? Can be both, no problem to me since it's in debug output.

**Comment by aguschin** [comment]
Ah, I see. @mike0sv, please disregard my comment above.

@omesser, could you please fetch upstream for your repo?
image
https://github.com/omesser/mlem/tree/verbose_log_obj_loads

File: mlem/contrib/pandas.py (hunk 1 above)
Comment (omesser) [reply mlem/contrib/pandas.py:696]: I don't know if there's a 1-size fits all honestly. I've tried both, decided to go with describe to be more friendly for large dataframes, but whatever you think @iterative/mlem  (or of course we can scratch this if you have a better idea / experience of debugging user-issues with mlem

Comment (mike0sv) [reply mlem/contrib/pandas.py:704]: Can we make this even lazier? I mean df.head is still executed even if log level is info. maybe there is something like

```python
a = lazy_str(df.head)
str(a) == ... # actual execution happens here
```

Comment (aguschin) [reply mlem/contrib/pandas.py]: There is a truncation applied to columns by default:
image

Comment (omesser) [reply mlem/contrib/pandas.py]: Just tested it with mnist as well 😄 it's safe (truncates columns like @aguschin mentioned). I think this should be controllable via pandas display options, the default is adaptive to display/terminal size

**Review by mike0sv — APPROVED** [review]

**Pull request merged**